### PR TITLE
Introduce CheckNonZeroSender

### PR DESCRIPTION
**Pull request by gavofyork**
skip check-dependent-cumulus

### CODE REVIEW

File: frame/system/src/extensions/check_non_zero_sender.rs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+	#[test]
+	fn zero_account_ban_works() {
+		new_test_ext().execute_with(|| {
+			let info = DispatchInfo::default();
+			let len = 0_usize;
+			assert_noop!(
+				CheckNonZeroSender::<Test>::new().validate(&0, CALL, &info, len),
+				InvalidTransaction::BadSigner
+			);
+			assert_ok!(CheckNonZeroSender::<Test>::new().validate(&1, CALL, &info, len),);
```
Comment (bkchr) [inline frame/system/src/extensions/check_non_zero_sender.rs:97]:
```suggestion
			assert_ok!(CheckNonZeroSender::<Test>::new().validate(&1, CALL, &info, len));
```

File: frame/system/src/extensions/check_non_zero_sender.rs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+impl<T: Config + Send + Sync> SignedExtension for CheckNonZeroSender<T>
+where
+	T::Call: Dispatchable<Info = DispatchInfo>,
+{
+	type AccountId = T::AccountId;
+	type Call = T::Call;
+	type AdditionalSigned = ();
+	type Pre = ();
+	const IDENTIFIER: &'static str = "CheckNonZero";
```
Comment (bkchr) [inline frame/system/src/extensions/check_non_zero_sender.rs:62]:
```suggestion
	const IDENTIFIER: &'static str = "CheckNonZeroSender";
```

File: primitives/runtime/src/transaction_validity.rs [hunk 3]
```diff
@@ -79,6 +79,8 @@ pub enum InvalidTransaction {
 	/// A transaction with a mandatory dispatch. This is invalid; only inherent extrinsics are
 	/// allowed to have mandatory dispatches.
 	MandatoryDispatch,
+	/// The sending address is disabled or known to be invalid.
+	BadSigner,
```
Comment (bkchr) [inline primitives/runtime/src/transaction_validity.rs:83]: This will require that clients update before.

**Review by bkchr — APPROVED** [review]

File: primitives/runtime/src/transaction_validity.rs (hunk 3 above)
Comment (gavofyork) [reply primitives/runtime/src/transaction_validity.rs:83]: :-(

Comment (bkchr) [reply primitives/runtime/src/transaction_validity.rs:83]: I mean it is not a hard requirement, aka we don't break authoring. We will just get a failure on `validate_block` when trying to decode the return value.

**Review by shawntabrizi — APPROVED** [review]

**Review by kianenigma — APPROVED** [review]
LGTM

**Pull request merged**